src/commands/git-metadata/__tests__/gitdb.test.ts

```diff
@@ -376,20 +467,131 @@ describe('gitdb', () => {
           output: [{name: 'origin', refs: {push: 'https://github.com/DataDog/datadog-ci'}}],
         },
       ],
-      // throw an exception after the update shallow to shortcut the rest of the test as we only
-      // care about updating the shallow clone, not about the rest of the process for this test
-      log: [{input: undefined, output: testError}],
-      raw: [],
-      revparse: [{input: '--is-shallow-repository', output: 'true'}],
+      log: [
+        {
+          input: ['-n 1000', '--since="1 month ago"'],
+          output: {
+            all: [
+              {
+                hash: '87ce64f636853fbebc05edfcefe9cccc28a7968b',
+              },
+              {
+                hash: 'cc424c261da5e261b76d982d5d361a023556e2aa',
+              },
+            ],
+          },
+        },
+      ],
+      // throw an exception after the failed unshallow to shortcut the rest of the test
+      raw: [{input: undefined, output: testError}],
+      revparse: [],
       version: [{input: undefined, output: oldGitVersion}],
       execSync: [],
-      axios: [],
+      axios: [
+        {
+          input: {
+            url: '/api/v2/git/repository/search_commits',
+            data: {
+              meta: {
+                repository_url: 'https://github.com/DataDog/datadog-ci',
+              },
+              data: [
+                {
+                  id: '87ce64f636853fbebc05edfcefe9cccc28a7968b',
+                  type: 'commit',
+                },
+                {
+                  id: 'cc424c261da5e261b76d982d5d361a023556e2aa',
+                  type: 'commit',
+                },
+              ],
+            },
+          },
+          output: {data: {data: []}},
+        },
+      ],
     })
     const upload = uploadToGitDB(logger, request, mocks.simpleGit as any, false)
     await expect(upload).rejects.toThrow(testError)
     mocks.expectCalls()
   })
 
+  test('should not unshallow repository if backend has all commits already', async () => {
+    const mocks = new MockAll({
+      getConfig: [
+        {
+          input: 'clone.defaultRemoteName',
+          output: defaultRemoteNameNotConfigured,
+        },
+      ],
+      fetch: [],
+      getRemotes: [
+        {
+          input: undefined,
+          output: [{name: 'origin', refs: {push: 'https://github.com/DataDog/datadog-ci'}}],
+        },
+      ],
+      log: [
+        {
+          input: ['-n 1000', '--since="1 month ago"'],
+          output: {
+            all: [
+              {
+                hash: '87ce64f636853fbebc05edfcefe9cccc28a7968b',
+              },
+              {
+                hash: 'cc424c261da5e261b76d982d5d361a023556e2aa',
+              },
+            ],
+          },
+        },
+      ],
+      raw: [],
+      revparse: [],
+      version: [],
```

Comment:
ℹ️ no revparse or version checks because we don't even attempt to unshallow (or check shallowness)